docs/RAG_serving_demo.md

```diff
@@ -0,0 +1,155 @@
+      name: 'ovms:py'
+  resources: {}
+  strategy:
+    type: Docker
+    dockerStrategy:
+      dockerfilePath: Dockerfile.redhat
+      buildArgs:
+        - name: IMAGE_NAME
+          value: 'registry.connect.redhat.com/intel/openvino-model-server@sha256:b8721a65da98bd9b354680ecb2dca8a4d60a1097babe9822ab264f8e951190f7'
+  postCommit: {}
+  source:
+    type: Git
+    git:
+      uri: 'https://github.com/openvinotoolkit/model_server'
+      ref: main
+    contextDir: demos/python_demos
+```
+It will trigger the build based on provided above Dockerfile and the based image. That image will be automatically pushed to the local cluster container registry in the OpenShift.
+
+## Adding the RAG servable configuration to a configmap
+
+The Model Server configuration including servable configuration needs to be attached to deployed pods. In this demo they will be passed via a configmap which can stored arbitrary files. The operator automatically mounts the contnent of a configmap to the serving pod.
+
+```
+pushd $(pwd)
+cd model_server/demos/python_demos/rag_chatbot/servable_stream
+echo "https://gist.githubusercontent.com/ryanloney/42b8ebe29f95ebd4382ee0b2bb50bea2/raw/cfbb679fefb6babec675c7806254a5fff29a5e6b/aipc.txt" > docs.txt
+oc create configmap rag-demo --from-file=config.json=config.json --from-file=graph.pbtxt=graph.pbtxt --from-file=model.py=model.py \
+--from-file=config.py=config.py --from-file=ov_embedding_model.py=ov_embedding_model.py \
+--from-file=ov_llm_model.py=ov_llm_model.py --from-file=docs.txt=docs.txt
+popd
+```
+
+## Adding Hugging Faces token to a secret and the model name
+The RAG demo can work with several models and it can be defined as an environment variable passed to the pod. Some models might require a token to authorize the model pull from Hugging Faces hub. Extra environment variables can be added to the deployment via a config_map or a secret
+
+```
+oc create secret generic rag-env --from-literal=HF_TOKEN=hf_GFb...
+oc create configmap rag-env --from-literal=SELECTED_MODEL=tiny-llama-2-chat-7b
+```
+
+## Deploying the service
+
+Now we can deploy the Model Server by applying the custom resource:
+```
+kubectl apply -f - <<EOF
+apiVersion: intel.com/v1alpha1
+kind: ModelServer
+metadata:
+  name: ovms-rag
+spec:
+  image_name: registry.toolbox.iotg.sclab.intel.com/cpp/model_server-gpu:py
+  deployment_parameters:
+    replicas: 1
+    extra_envs_secret: "rag-env"
+    extra_envs_configmap: "rag-env"
+  models_settings:
+    single_model_mode: false
+    config_configmap_name: "rag-demo"
+  server_settings:
+    log_level: "INFO"
+  service_parameters:
+    grpc_port: 8080
+    rest_port: 8081
+EOF
+```
+It will deploy new pod and service `ovms-rag`. In the pod initialization, it will pull and load the LLM and embedding models from Hugging Faces. Then the documentes from provided docs.txt file fill be downloaded and analysed. The server will be updating the analysis when the docs.txt file is modified.
```

Comment:
```suggestion
It will deploy new pod and service `ovms-rag`. In the pod initialization, it will pull and load the LLM and embedding models from Hugging Face. Then the documents from provided docs.txt file will be downloaded and analyzed. The server will be updating the analysis when the docs.txt file is modified.
```